rskj-core/src/main/java/co/rsk/peg/PegoutCreationEntry.java

```diff
@@ -30,19 +30,25 @@ public boolean equals(Object o) {
         }
 
         PegoutCreationEntry that = (PegoutCreationEntry) o;
-        return getBtcTxHash().equals(that.getBtcTxHash()) && getRskTxHash().equals(that.getRskTxHash());
+
+        if (!getBtcTxHash().equals(that.getBtcTxHash())) {
+            return false;
+        }
+        return getRskTxHash() != null ? getRskTxHash().equals(that.getRskTxHash()) : that.getRskTxHash() == null;
     }
 
     @Override
     public int hashCode() {
-        return getBtcTxHash().hashCode() + getRskTxHash().hashCode();
+        int result = getBtcTxHash().hashCode();
+        result = 31 * result + (getRskTxHash() != null ? getRskTxHash().hashCode() : 0);
```

Comment:
It's a convention in Java to avoid hashcode collisions when there are nullable class members.

For more details take a look at: https://www.geeksforgeeks.org/why-does-javas-hashcode-in-string-use-31-as-a-multiplier/

Reply:
That's interesting. Thanks!